numpy/_core/shape_base.py

```diff
@@ -233,9 +233,12 @@ def vstack(tup, *, dtype=None, casting="same_kind"):
 
     Parameters
     ----------
-    tup : sequence of ndarrays
+    tup : ndarray or sequence of ndarrays
```

Comment:
I'd keep this as it was - an `ndarray` is not the usual input, it just happens to be treated as a sequence of ndarray as well.

Reply:
My logic was that the *expected* input is `sequence of ndarray` - a single ndarray just happens to be treated as a sequence as well, which is a side effect of it behaving like a `Sequence` but not really what one expects to put in. In other words, if one things of this as typing, it would be reasonable to warn a user about a single array input - the logical thing might be to explicitly pass in, e.g., `list(array)`.

Reply:
@mhvk are you suggesting to add a warning when a single array is passed as input?
I still think adding a note in the documentation could help users better understand this behavior. While it's logical, the consequence of passing a single array might not always be expected, so clarifying this could prevent confusion

Reply:
@Kh4L - I just think we should keep the description on this line `tup: sequence of ndarray` - but keep your note in the comment that a single `ndarray` is interpreted as a sequence too.

Reply:
```suggestion
    tup : sequence of ndarrays
```